src/builder-git.c

```diff
@@ -731,14 +763,17 @@ builder_git_shallow_mirror_ref (const char     *repo_location,
             "fetch", "--depth", "1", "origin", full_ref_colon_full_ref, NULL))
     return FALSE;
 
-  if (!git (mirror_dir, NULL, 0, error,
-            "lfs", "fetch", "origin", full_ref, NULL))
+  if (git_lfs_is_available ())
     {
-      g_autoptr(GError) log_error = NULL;
+      if (!git (mirror_dir, NULL, 0, error,
```

Comment:
This is basically just `if (git_lfs_is_available () && !git (...))`. Removes more nesting; could be done everywhere in this commit.